src/js/state.js

```diff
@@ -0,0 +1,77 @@
+class State {
+	/**
+	* Class constructor.
+	* @param {RadioNodeList} [inputs] - A NodeList of radio input elements
+	*/
+	constructor(inputs) {
+		this.inputs = inputs;
+		this.parent = inputs.length > 1 ? this.inputs[0].closest('.o-forms-input') : null;
+
+		this._verify();
+
+		this.stateEl = null;
+		this.className = {
+			saving: 'o-forms-input--saving',
+			saved: 'o-forms-input--saved'
+		};
+	}
+
+	/**
+	* State setter
+	* @param {String} [state] type of state to display
+	*/
+	set(state) {
+		if (state === 'saving') {
+			this._saving();
+		} else if (state === 'saved') {
+			this._saved();
+		} else if (state === 'none') {
+			this._remove();
+		}
+	}
+
+	/**
+	* Saving state
+	* @access private
+	*/
+	_saving() {
+		if (this.stateEl === null) {
+			this.stateEl = document.createElement('span');
+			this.stateEl.classList.add('o-forms-input__state');
+			this.parent.classList.add(this.className.saving);
+			this.parent.append(this.stateEl);
+		}
+	}
+
+	/**
+	* Saved state
+	* @access private
+	*/
+	_saved() {
+		this.parent.classList.replace(this.className.saving, this.className.saved);
```

Comment:
`classList.replace` does not exist in IE. We would need to add DOMTokenList to the browserFeatures.required array for `classList.replace` to get polyfilled. It would be good to run the tests across our browsers via `obt --browserstack` and ensure they all pass